File: app/models/dish_type.rb
class DishType < ActiveRecord::Base
  attr_accessible :name

  has_many :dishes, :inverse_of => :dish_type
  has_many :restaurants, :through => :dishes
  validates :name, :presence => true
  validates :name, :uniqueness => true

  def dish_type_id
  end
end

File: app/assets/javascripts/models/dish_type.js
DR.Models.DishType = Backbone.RelationalModel.extend({
	schema: {
		name: "Text"
	},

  urlRoot: "/dish_type",
  relations:[{
    type: Backbone.HasMany,
    key: "dishes",
    relatedModel: "DR.Models.Dish",
    collectionType: "DR.Collections.DishTypeDishes",

    reverseRelation: {
      key: "dishType",
      keySource: "dish_type_id",
      includeInJSON: "id"
    }
  },
  
  {
    type: Backbone.HasMany,
    key: "restaurants",
    relatedModel: "DR.Models.Restaurant",
    collectionType: "DR.Collections.DishTypeRestaurants",

    reverseRelation: {
      key: "dishType",
      keySource: "dish_type_id",
      includeInJSON: "id"
    }
  }

  ]


});